CSVSource/build.gradle

```
@@ -78,3 +78,31 @@ tasks.withType(Test) {
         systemProperty 'EntFullFilePath', rootProject.findProperty('EntFullFilePath')?:"b.csv"
     }
 }
+
+// Using gradle-dcompose-plugin to build CSVSource docker image
+dcompose {
+    if (rootProject.hasProperty('dockerRegistry')) {
+        def registryName = rootProject.getProperty('dockerRegistry')
+        csvSource {
+            buildFiles = project.copySpec {
+                from "$projectDir/src/main/docker/"
+                from "$projectDir/build/distributions/CSVSource.tar"
+            }
+            if (rootProject.hasProperty('csvImageTag')) {
+                repository = registryName + '/csv-source' + rootProject.getProperty('csvImageTag')
```

Something similar.  If you need the name in this format, you can mostly do something like

```suggestion
                repository = registryName + "${project.name - 'Source' + '-source}.toLowercase()" + rootProject.getProperty('imageTag')
```